### DISCUSSION TITLE
posts/2023/10/testing-testing/

### DISCUSSION THREAD

**Discussion by giscus** [brycewray/comments#37] [Announcements]
# posts/2023/10/testing-testing/

What I found in recent evaluations of web browsers and the new Cloudflare Fonts service.

https://www.brycewray.com/posts/2023/10/testing-testing/

**Comment by davidsneighbour** [comment]
I would like to be the troll in the last row and point something out:

[Quick MeMe for entertainment](https://pictr.com/image/BaJUKX)

What you describe above is basically Mac on an unknown RAM amount, then a subset of that RAM amount to run Windows and then a completely different Computer with maybe the same amount of RAM, but different processor. Speed tests should IMHO be done on the same hardware (even if it's simulated) to be comparable. When I had a 8GB RAM station with Ubuntu, Chromium was deadly slow. Now on a 64GB RAM station I have no complaints with around 40 tabs open at any given point (lots of tab groups). 

Maybe re-run your tests and build each OS in one of these virtual machines you tested Windows in. Give them all the same amount of RAM. Then that would be a comparable test between the OSSes. 

I think the reason people don't obsess too much about speed anymore for is exactly that reason - they have more RAM. 

(If that is comparable, I run the speedometer test on this browser (Chrome on Ubuntu with 2h worktime under the belt, a bunch of plugins that parse the current page (adblock, etc), and lots of tabs open, 64GB machine with i7 processor) and got a Arithmetic Mean:279 ± 24 (8.4%). Don't know what that means ;)

Chrome by the way has a new mode that you might need to enable first, where unfocused tabs after a while clean up and unload. After a while ;) 

That "after a while" too is important, because on Ubuntu I had the feeling (8GB machine) that Firefox was fast, but lost all speed after 1h and a bunch of tabs that were open and closed. 

Long story short: In the end it depends on how the browser you are using works with what it's given by the system.

**Comment by brycewray** [comment]
@davidsneighbour I have added the following information to the paragraph that began with “For each test...” — clarifying the lengths to which I went in order to ensure fairness within each testing environment. Of course, ***cross***-environmental evaluations (*e.g.*, ”Chrome on a souped-up 2023 Mac is [x] times faster than Firefox in emulated Windows 12 and [x] times faster than Brave on Linux”) definitely would have been “apples *vs.* oranges *vs.* lemons” and therefore of little value; but I strove to provide only “apples-*vs.*-apples” comparisons.

> Each test took place with the browser in a private/“incognito” window with all extensions disabled, with no other tabs or other browsers open, and with all content filtering otherwise turned off; also, on the Windows emulation, I used the same Parallels Desktop *profile* across the browser testing. Thus, on each separate environment, the tests constituted as fair an “apples-*vs.*-apples” comparison as I could possibly arrange. (Obviously, the native macOS experience on the Mac Studio was *far* faster than either the Windows 11 emulation on the Mac studio or the native Linux experience on the older, slower iMac; but I believe that, **within each environment**, the comparisons were fair and indicative of what each browser could do.)
